Make editor work with current version of the game

girtonman opened the pull request:
After doing some troubleshooting, it looks like the main issue that was causing it to crash was that the seed for the save was too big for a js float, so it was being rounded. 
For example:
```
  "seed": 1461819000922140955,
```
was being rounded to
```
  "seed": 1461819000922141000,
```

This would cause the game to crash when the map gen for the rounded seed would conflict with the map position info in the save file.

Adding `lossless-json` solves this issue pretty easily. I did some very simple testing and the tool seems to be working again with this fix.

girtonman committed:
Address eslint warnings

girtonman committed:
Add lossless-json package to fix a rounding error

girtonman commented:
Closes the following issues:
#25 #27 #28

zuelong commented:
Wow!  Thanks so much for digging and looking into this!  I'll be sure to get this deployed and maybe add some CI so I can stop doing manual deploys 😅

zuelong commented on src/components/Advanced.js:
```diff
@@ -1,5 +1,6 @@
 import React, { Component } from 'react';
 import { connect } from 'react-redux';
+const LosslessJSON = require('lossless-json');
```
I just realized this won't actually work without proper import syntax, can you update this please?

girtonman replied:
I'm pretty new to this tech stack. I tried `import LosslessJSON from 'lossless-json';` and it was throwing null errors.

I'm open to suggestions.

girtonman replied:
Read up on ES6 modules. Figured out what I needed to do based on the exports for `lossless-json`.

zuelong commented on src/components/Advanced.js:
```diff
@@ -44,6 +45,7 @@ class Advanced extends Component {
     encrypt = () => {
         try {
             const decrypted = this.state.json;
+            console.log(decrypted);
```
you'll probably want to remove this 😅

girtonman replied:
I noticed that while doing some other work on it today. Oops!

I'll remove it.

girtonman replied:
Fixed

zuelong commented on src/components/FileUpload.js:
```diff
@@ -33,7 +34,8 @@ class FileUpload extends Component {
         try {
             let decrypted = this.xorWithKey(atob(encrypted), "key");
-            return JSON.parse(decrypted);
+            console.log(decrypted);
```
You'll probably want to remove this from here as well.

girtonman replied:
Same as above :)

I'll remove this one as well.

girtonman replied:
Fixed.

zuelong reviewed (changes requested):
Just a few suggestions before I can merge this in, but great work!

I'll get it tested and merged after these are corrected

girtonman commented:
I've also been messing around with it and found a couple minor changes that need to be made to handle the new LosslessNumber type.

Throughout a day of testing and using it, I'm feeling even more confident in this fix.

I'm in the process of adding all of the new cards/relics to `CardsJSON` and `RelicsJSON`. Would you like me to shoe-horn them into this PR or make a new one for that? There are about 90 cards that need to be added. 😮 I haven't figured out how many relics yet though, but that should _hopefully_ be a shorter list.

girtonman commented:
Resolved all feedback and also made a small fix to handle the fact that the value of `upgrades` can be a number of a `LosslessValue` depending on its source (file upload or card selector),

girtonman commented:
Also moved the changes I was talking about for `CardsJSON` and `RelicsJSON` to another branch that I can make a PR for once this one is wrapper up.

girtonman committed:
Change requires to imports

girtonman committed:
Fix issue with adding cards

Since `upgrades` can be a `LosslessValue` or a number, the ternary condition needs to handle both.

girtonman committed:
Update repo to get ready for gh-pages deploys

girtonman committed:
Merge pull request #1 from girtonman/feature/gh-pages-init

feature/gh-pages-init
